agent/internal/backend/api/api.go

```diff
@@ -829,6 +829,40 @@ func NewBlockedUserEventProperties_OutputFromFace(that BlockedUserEventPropertie
 	return this
 }
 
+type RedirectedIPEventProperties struct {
+	ActionId string                            `json:"action_id,omitempty"`
+	Output   RedirectedIPEventPropertiesOutput `json:"output"`
+}
+
+type RedirectedIPEventPropertiesOutput struct {
+	IpAddress string `json:"ip_address"`
+	URL       string `json:"url"`
+}
+
+func NewRedirectedIPEventPropertiesFromFace(that RedirectedIPEventPropertiesFace) *RedirectedIPEventProperties {
+	return &RedirectedIPEventProperties{
+		ActionId: that.GetActionId(),
+		Output:   that.GetOutput(),
+	}
+}
+
+type RedirectedIPEventPropertiesFace interface {
+	GetActionId() string
+	GetOutput() RedirectedIPEventPropertiesOutput
+}
+
+type RedirectedIPEventPropertiesOutputFace interface {
+	GetIpAddress() string
+	GetURL() string
+}
+
+func NewRedirectedIPEventPropertiesOutputFromFace(that RedirectedIPEventPropertiesOutputFace) *RedirectedIPEventPropertiesOutput {
```

Comment:
I don't like the `that` naming convention. I'd rather have it shortened but meaningful, like `face`, `event`, or whatever, which is much more readable just below. (This comment is valid for some other places)

Reply:
Agreed - that was generated and that will be rewritten at some point.